modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/utils/APIImportUtil.java

```diff
@@ -368,6 +368,158 @@ public static void importAPI(String pathToArchive, String currentUser, boolean i
 
     }
 
+    /**
+     * This method updates an existing API. The state of the API is preserved in the update process
+     *
+     * @param apiID                    UUID of the API to be updated
+     * @param pathToArchive            Archive used to update the API
+     * @param currentUser              User for update operation
+     * @param isDefaultProviderAllowed Is default provider allowed to do the operation
+     * @throws APIImportException If there is an error the exception is thrown
+     */
+    public static void updateAPI(String apiID, String pathToArchive, String currentUser, boolean isDefaultProviderAllowed)
+            throws APIImportException {
+
+        API importedApi = null;
+        API targetApi = null;
+        String prevProvider;
+        APIDefinition definitionFromOpenAPISpec = new APIDefinitionFromOpenAPISpec();
+
+        String pathToJSONFile = pathToArchive + APIImportExportConstants.JSON_FILE_LOCATION;
+
+        try {
+            String jsonContent = FileUtils.readFileToString(new File(pathToJSONFile));
+            JsonElement configElement = new JsonParser().parse(jsonContent);
+            JsonObject configObject = configElement.getAsJsonObject();
+
+            //locate the "providerName" within the "id" and set it as the current user
+            JsonObject apiId = configObject.getAsJsonObject(APIImportExportConstants.ID_ELEMENT);
+            prevProvider = apiId.get(APIImportExportConstants.PROVIDER_ELEMENT).getAsString();
+            String prevTenantDomain = MultitenantUtils
+                    .getTenantDomain(APIUtil.replaceEmailDomainBack(prevProvider));
+            String currentTenantDomain = MultitenantUtils
+                    .getTenantDomain(APIUtil.replaceEmailDomainBack(currentUser));
+
+            // If the original provider is preserved,
+            if (isDefaultProviderAllowed) {
+                if (!StringUtils.equals(prevTenantDomain, currentTenantDomain)) {
+                    String errorMessage = "Tenant mismatch! Please enable preserveProvider property " +
+                            "for cross tenant API Import ";
+                    log.error(errorMessage);
+                    throw new APIImportException(errorMessage);
```

Comment:
no need to throw and log the same error twice.

Reply:
will refactor